### PR TITLE
Add options to specify raiden client executable to use

### PR DESCRIPTION
**Pull request by palango**
Fixes https://github.com/raiden-network/scenario-player/issues/593

This PR removes the hardcoded `raiden_version` in the scenario file and replaces it with options to set the executable to use from the CLI, the environment file or overwrite on a per-node basis in the scenario.

This also makes it possible to remove the `RaidenReleaseKeeper` class. As `flake8` complained about some code this also fixes some warnings not strictly related to the functionality.

### CODE REVIEW

File: scenario_player/main.py [hunk 1]
```diff
@@ -323,6 +328,9 @@ def run_(
             )
         notify_tasks_callable = post_task_state_to_rc
 
+    # TODO: make this used
+    print(notify_tasks_callable)
```
Comment (fredo) [inline scenario_player/main.py:332]: is this intended to stay?

File: scenario_player/node_support.py [hunk 2]
```diff
@@ -68,109 +56,16 @@
...
-    def __init__(self, runner: "ScenarioRunner", index: int, raiden_version, options: dict):
+    def __init__(self, runner: "ScenarioRunner", index: int, raiden_client: str, options: dict):
         self._runner = runner
         self._index = index
-        self._raiden_version = raiden_version
         self._options = options
         self._nursery: Optional[Nursery] = None
+
+        # Overwrite raiden client to use, if given in options
+        self._raiden_client = options.pop("raiden-client", raiden_client)
```
Comment (fredo) [inline scenario_player/node_support.py:67]: Just to assure, is option["raiden-client"] used somewhere afterwards? As it pops it from the dict.

**Review by fredo — COMMENTED** [review]
In general looks good to me, left some remarks. 

One question pops up. When using the executable, if we would like to show the current version we could query it from the CLI, right? I saw that some logs showing the version used were removed. Would be nice to see it somewhere.

File: scenario_player/node_support.py (hunk 2 above)
Comment (palango) [reply scenario_player/node_support.py:67]: No, it isn't

File: scenario_player/main.py (hunk 1 above)
Comment (palango) [reply scenario_player/main.py:332]: It was passed to `orchestrate` before, but never used afterwards. Maybe @ulope can shine some light on this?

File: examples/scenario-example-v2.yaml [hunk 3]
```diff
@@ -72,6 +69,7 @@ nodes:
   #node_options:
   #  0:
   #    matrix-server: https://transport01.raiden.network
+  #    raiden-client: light-client
```
Comment (ulope) [inline examples/scenario-example-v2.yaml:72]: Hm mixing in this "meta" option into the cli options feels a bit "unclean".

File: scenario_player/node_support.py [hunk 4]
```diff
@@ -516,7 +417,6 @@ def __init__(
         if config.restore_snapshot:
             if self.snapshot_manager.restore():
                 self.snapshot_restored = True
-        log.info("Using Raiden version", version=config.raiden_version)
```
Comment (ulope) [inline scenario_player/node_support.py:519]: Maybe keep this and log the full path to the executable that is being used?

**Review by ulope — APPROVED** [review]
LGTM.

I'm still a bit sad that we lose the ability to run different versions without external tooling.
But nobody ever using that feature speaks pretty clearly that it wasn't very useful.

File: scenario_player/main.py (hunk 1 above)
Comment (ulope) [reply scenario_player/main.py:332]: The idea of that callback was to have a way of notifying external tools (e.g. rocketchat) about the task state. It's very possible that this got broken in one of the refactorings along the way.

File: examples/scenario-example-v2.yaml (hunk 3 above)
Comment (palango) [reply examples/scenario-example-v2.yaml:72]: I know what you mean. Any other ideas? 

I'd also be ok with not having this feature at this point and implementing a "proper" solution later.

File: scenario_player/main.py (hunk 1 above)
Comment (palango) [reply scenario_player/main.py:332]: Ok, then I create an issue to bring it back when we need it

**Pull request merged**